Rock.Messaging.SQS/MQ/SQSQueueReceiver.cs

```diff
@@ -103,71 +106,83 @@ private void DoStuff()
                     continue;
                 }
 
-                foreach (var message in response.Messages)
+                if (_parallelHandling)
                 {
-                    if (_stopped)
+                    Parallel.ForEach(response.Messages, Handle);
+                }
+                else
+                {
+                    foreach (var message in response.Messages)
                     {
-                        break;
+                        Handle(message);
                     }
+                }
+            }
+        }
+
+        private void Handle(Message message)
+        {
+            if (_stopped)
+            {
+                return;
+            }
 
-                    var handler = MessageReceived;
+            var handler = MessageReceived;
+
+            if (handler != null)
+            {
+                var receiptHandle = message.ReceiptHandle;
 
-                    if (handler != null)
+                Action acknowledge =
+                    () =>
                     {
-                        var receiptHandle = message.ReceiptHandle;
+                        Exception deleteException = null;
+                        DeleteMessageResponse deleteResponse = null;
 
-                        Action acknowledge =
-                            () =>
+                        for (int i = 0; i < 3; i++)
+                        {
+                            try
                             {
-                                Exception deleteException = null;
-                                DeleteMessageResponse deleteResponse = null;
+                                deleteException = null;
+                                deleteResponse = null;
```

Comment:
Seems we do not need to set this to null given the next line will set the value when the .OverAsync returns.  Am I missing something?

Reply:
Not sure, this was also just moved around.  @bfriesen thoughts?

Reply:
Took me a few minutes to remember, but this is to support better error reporting. Here's the hypothetical scenario:

We're in the retry loop, trying up to three times. The first time around, it isn't successful (doesn't matter why). The second time around, calling `_sqs.DeleteMessageAsync` does not throw an exception, but the response does not have a status code of `OK`. Note that `deleteResponse` *has* a value at this point before we loop again. On the third time around, `_sqs.DeleteMessageAsync` throws an exception, and it never assigns to `deleteResponse`. If assigning to null doesn't happen immediately before, then the error we report has mismatched error information and would be confusing - the exception would describe what went wrong when making the call to SQS, while the additional information from `deleteResponse` would describe a successful call to SQS with an unsuccessful result.